server/handler/base.go

```diff
@@ -391,6 +401,31 @@ func selectionToReviewersRequest(s reviewer.Selection) github.ReviewersRequest {
 	return req
 }
 
+func (b *Base) dismissStaleReviewsForResult(ctx context.Context, prctx pull.Context, result common.Result) error {
+	if len(result.DiscardedReviews) == 0 {
+		return nil
+	}
+
+	for _, d := range result.DiscardedReviews {
```

Comment:
`common.Result` is a tree that follows the structure of the policy, but `dismissStaleReviewsForResult` looks like it is called on the root only. I think this is missing two steps:

1. Aggregate the discarded reviews from all children into a single list to process
2. Deduplicating that list so that reviews that are discarded by multiple rules are only dismissed once

Reply:
gotcha, I was thinking that result was a 1:1 to a rule approval. so I thought I was aggregating the discarded approvals per rule inside of common.Result.DiscardedReviews, but i'll look deeper into understanding the structure of common.Result.